discovery/aws/ec2.go

```diff
@@ -112,7 +112,7 @@ func (*EC2SDConfig) Name() string { return "ec2" }
 
 // NewDiscoverer returns a Discoverer for the EC2 Config.
 func (c *EC2SDConfig) NewDiscoverer(opts discovery.DiscovererOptions) (discovery.Discoverer, error) {
-	return NewEC2Discovery(c, opts.Logger, opts.Metrics)
+	return NewEC2Discovery(c, opts.Logger, opts.Metrics, opts.ConfigName)
```

Comment:
I think, where you find yourself editing 50 of these to add one more parameter, it might be better to just pass `opts`, then we won't have to do this again.
Do that in a separate commit described as a refactor, to make it more obvious what happened.